src/Uri.php

```diff
@@ -645,7 +645,7 @@ private static function getFilteredQueryString(UriInterface $uri, array $keys)
      *
      * @return string
      */
-    private static function generateQueryString($key, $value)
+    private static function generateQueryString(string $key, ?string $value): string
```

Comment:
you need to call this method from `withQueryValues` with explicit casting because numeric array keys are set as integer and the value might also be an int. since it's hard to phpdoc array element/key types, it's better to support string-castable contents.

```
foreach ($keyValueArray as $key => $value) {
    $result[] = self::generateQueryString((string) $key, (string) $value);
}
```